indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -743,7 +743,8 @@ public void onFailure(Throwable t)
           }
 
           if (System.currentTimeMillis() > nextCheckpointTime) {
-            sequenceToCheckpoint = getLastSequenceMetadata();
```

Comment:
Can sequenceToCheckpoint be non-null before this point?
If yes, perhaps you could add the `isCheckpointed` condition to the if condition, rather than changing the body in the if block.

Reply:
Yes, the sequenceToCheckpoint can be non-null when isPushRequired becomes true. 
I'll isCheckpointed condition to the if condition.